proposals/0055-csp.md

```diff
@@ -0,0 +1,251 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+**If you have feedback and the feature is released as experimental, please leave it on the Stage 3 PR. Otherwise, comment on the Stage 2 issue (links below).**
+
+- Start Date: 2025-05-13
+- Reference Issues: 
+- Implementation PR: 
+- Stage 2 Issue: https://github.com/withastro/roadmap/issues/1149
+- Stage 3 PR: https://github.com/withastro/roadmap/discussions/377
+
+# Summary
+
+Add a new feature that allows Astro assets, mainly scripts and styles, to support [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP).
+
+# Example
+
+```js
+// astro.config.mjs
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+    security: {
+        csp: true
+    }
+})
+```
+
+# Background & Motivation
+
+Include any useful background detail that that explains why this RFC is important.
+What are the problems that this RFC sets out to solve? Why now? Be brief!
+
+It can be useful to illustrate your RFC as a user problem in this section.
+(ex: "Users have reported that it is difficult to do X in Astro today.")
+
+# Goals
+
+- Support at least one solution (`nonce`, hash, etc.) that would support CSP.
+- Allow Astro scripts (client islands, server islands, view transitions, etc.) and Astro styles to work out of the box.
+- Update adapters to support the feature, if needed.
+- Make the feature opt-in, even when it's outside experimental.
+- Support the [dev server](#dev-server) (stretch)
+- View transitions (limited/opt-in support)
+
+# Non-Goals
+
+- Provide *multiple* solutions to fix the same problem
+- Support for third-party scripts dynamically imported by users e.g. `document.createElement("script")`, `import("https://path.to/script.js")`;
+- Support for third-party styles dynamically imported by users.
```

Comment:
If I undestand correctly, `injectCspAsset` would allow this

Reply:
I want to rephrase it then. What I meant is that **out of the box** Astro won't create a hash for the dynamically imported script/style. Any suggestions?

Reply:
```suggestion
- Automatic support for third-party styles dynamically imported by users.
```